### ISSUE TITLE
Swipe gestures in the overlay

### ISSUE THREAD

**Issue by hjo** [ephread/Instructions#135]
In my app, I had some instructions to teach the user some gestures they could use in the app - i.e. swiping up does this, swiping down does that, swiping left does this other thing and swiping right does that other thing.

In user testing I found that when users got those instructions they wouldn't tap - instead they would try to swipe on the overlay as instructed. To give them a better experience, I added gesture recognizers to the CoachMarksController overlay so that I could add actions to detect these events which would move the user to the next instruction (in past versions of Instructions, overlay was just a view so this was easy to do).

I've now upgraded to the latest version of Instructions where overlay has become an OverlayManager and there is no longer access to the underlying overlay view. I couldn't figure out an alternative way to implement the functionality I had before. Do you have any ideas of how this could be done, or alternatively perhaps OverlayManager should be enhanced so that the delegate can get the various swipe events?

**Comment by ephread** [comment]
Hey @hjo, I didn't think of that case. Given your needs are very specific, I think I'll just make the overlay view public again in the next version so that you'll be able to attach your gesture recognizers again!

**Comment by hjo** [comment]
Thanks - that would be great!

**Comment by ephread** [comment]
In 1.2.0, you can [configure ornaments](https://github.com/ephread/Instructions#customizing-ornaments-of-the-overlay).

I know it feels a bit hacky but `overlay.superview` is the view to which you should attach your gesture recognizer. In the future, I think I'll create a specific delegate which will handle the configuration of the overlay (gesture recognizer, custom animation, etc.), to make it more flexible.

**Comment by hjo** [comment]
Unfortunately the overlayView is still not public even in 1.2.0?

**Comment by ephread** [comment]
That's correct. After putting more thoughts into this, I don't want to expose `OverlayView`. The reasons are simple, the less I expose the internals, the less I have to be careful about backward compatibility when reworking them.

I honestly admit that I released 1.2.0 and forgot about this long lasting issue. I'm sorry for that.

That said, my previous comment holds true. The `OverlayView` is exposed as an `UIView` via `overlay`'s superview, but this `overlay` here is not `CoachMarksController.overlay`, it's the parameter passed to:

```swift
func coachMarksController(_ coachMarksController: CoachMarksController,
                          configureOrnamentsOfOverlay overlay: UIView)
```

In 1.3.0, I'm going to add a `UIView` property to `CoachMarksController.overlay`, simply called `view`. That'll enable you to add your gesture recognizer without exposing the internals.
